### PR TITLE
Removed activation from Contrib Epoch Metrics, updated tests

### LINKED ISSUE #404: Remove `activation` argument from AveragePrecision and ROC_AUC
I propose to remove `activation` argument from `AveragePrecision` and `ROC_AUC`:
https://github.com/pytorch/ignite/blob/862ab0073e461a32bf5b53fa015f88c143ae9079/ignite/contrib/metrics/roc_auc.py#L23-L28
to be coherent with `Accuracy` et friends and the usage of `output_transform` to apply the binarization/activation :
https://github.com/pytorch/ignite/blob/862ab0073e461a32bf5b53fa015f88c143ae9079/ignite/metrics/accuracy.py#L81-L93

### PR DESCRIPTION
**Pull request by anmolsjoshi**
Fixes #404 

Description:
Removed activation parameter from ROC_AUC, AveragePrecision, added doctstring that activation must be added using output_transform. 

Removed existing activation tests, refactored output_transform tests to include activation. 

Check list:
* [x] New tests are added (if a new feature is modified)
* [x] New doc strings: text and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by anmolsjoshi** [comment]
@vfdev-5 please review

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 1]
```diff
@@ -51,35 +51,47 @@ def test_ap_score_2():
...
+
+    def update_fn(engine, batch):
+        idx = (engine.state.iteration - 1) * batch_size
+        y_true_batch = np_y[idx:idx + batch_size]
+        y_pred_batch = np_y_pred[idx:idx + batch_size]
+        return idx, torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+
+    engine = Engine(update_fn)
+
+    roc_auc_metric = AveragePrecision(output_transform=lambda x: (x[1], x[2]))
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py]:
```suggestion
    ap_metric = AveragePrecision(output_transform=lambda x: (x[1], x[2]))
```

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 2]
```diff
@@ -51,35 +51,47 @@ def test_ap_score_2():
...
+    roc_auc_metric.attach(engine, 'ap')
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py]:
```suggestion
    ap_metric.attach(engine, 'ap')
```

File: tests/ignite/contrib/metrics/test_roc_auc.py [hunk 3]
```diff
@@ -52,36 +52,17 @@ def test_roc_auc_score_2():
     assert roc_auc == np_roc_auc
 
 
-def test_roc_auc_score_with_activation():
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_roc_auc.py]: May aks you why did you removed this test ?

File: tests/ignite/contrib/metrics/test_average_precision.py (hunk 2 above)
Comment (anmolsjoshi) [reply tests/ignite/contrib/metrics/test_average_precision.py]: Incorporated!

File: tests/ignite/contrib/metrics/test_average_precision.py (hunk 1 above)
Comment (anmolsjoshi) [reply tests/ignite/contrib/metrics/test_average_precision.py]: Incorporated!

**Comment by anmolsjoshi** [comment]
@vfdev-5 thanks for the review, comments have been incorporated. please review and merge!

File: ignite/contrib/metrics/average_precision.py [hunk 4]
```diff
@@ -21,14 +19,23 @@ class AveragePrecision(EpochMetric):
...
+            y_pred = torch.softmax(y_pred)
+            return y_pred, y
+
+        avg_precision = AveragePrecision(activated_output_transform)
+
     """
     def __init__(self, activation=None, output_transform=lambda x: x):
-        super(AveragePrecision, self).__init__(partial(average_precision_compute_fn, activation=activation),
-                                               output_transform=output_transform)
+        super(AveragePrecision, self).__init__(partial(average_precision_compute_fn), output_transform=output_transform)
```
Comment (vfdev-5) [inline ignite/contrib/metrics/average_precision.py]: Please remove `partial` and the import

File: ignite/contrib/metrics/roc_auc.py [hunk 5]
```diff
@@ -22,14 +20,23 @@ class ROC_AUC(EpochMetric):
...
+            return y_pred, y
+
+        roc_auc = ROC_AUC(activated_output_transform)
+
     """
-    def __init__(self, activation=None, output_transform=lambda x: x):
-        super(ROC_AUC, self).__init__(partial(roc_auc_compute_fn, activation=activation),
-                                      output_transform=output_transform)
+    def __init__(self, output_transform=lambda x: x):
+        super(ROC_AUC, self).__init__(partial(roc_auc_compute_fn), output_transform=output_transform)
```
Comment (vfdev-5) [inline ignite/contrib/metrics/roc_auc.py]: Same here for `partial`

File: ignite/contrib/metrics/average_precision.py (hunk 4 above)
Comment (anmolsjoshi) [reply ignite/contrib/metrics/average_precision.py]: Incorporated!

File: ignite/contrib/metrics/roc_auc.py (hunk 5 above)
Comment (anmolsjoshi) [reply ignite/contrib/metrics/roc_auc.py]: Incorporated!

**Comment by anmolsjoshi** [comment]
@vfdev-5 currently both test have a binary case, is there any value added in adding a multiclass case?

**Comment by vfdev-5** [comment]
> currently both test have a binary case, is there any value added in adding a test for multiclass case?

@anmolsjoshi I'm not sure that `roc_auc_score` or `average_precision_score` support multiclass case.

**Comment by vfdev-5** [comment]
@anmolsjoshi yes, but pay attention on the form of `y_true`. It can not be something like `[0, 1, 2, 3, 4]`, it can be some binary class indicator...

Anyway, I'm not sure that this will add any value as we test sklearn functions and not ignite ones.

**Comment by anmolsjoshi** [comment]
@vfdev-5 understood, my bad. the docs state this is for binary and multilabel cases only, so multiclass should be ignored! 

any thing else needed for this PR?

**Comment by vfdev-5** [comment]
@jasonkriss please review and merge if OK

**Comment by jasonkriss** [comment]
Great idea on this guys. LGTM.
